service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -149,6 +152,12 @@ func Invoke(
 	)
 
 	if err != nil {
+		var deserializationError *serialization.DeserializationError
+		var serializationError *serialization.SerializationError
+		// convert serialization errors to be captured as serviceerrors across gRPC calls
```

Comment:
it probably make more sense to do this in a grpc interceptor then we will have it for all apis. Can be done later. Or maybe just define them as grpc errors.

Reply:
I am also ok with removing those definitions and always use DataLoss. I checked the codebase quickly and I don't see them getting used outside the persistence layer.

Reply:
I completely agree on removing these errors and just catching DataLoss errors. However, I double-checked the code base too and found out that Deserialization errors are being used outside of the persistence layer (for example -`common/nexus/payload_serializer`)

No harm in changing the error returned while un-marshalling JSON to dataloss, imo, but I wanted to double check before authoring this change. 

tldr - there are places where deserialization errors are being used across non-RPC domains. Do we still want to eradicate them completely?

Reply:
I personally like the idea of catching these errors in the interceptor and then converting them to a dataloss. Shall be going ahead with that.